### PR TITLE
[llfio,ned14-internal-quickcpplib,outcome,status-code] Update to 2023-11-06

### PR DESCRIPTION
**Pull request by BurningEnlightenment**
The ports are tightly coupled and the new versions contain breaking changes which is why they can't be updated separately.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/ned14-internal-quickcpplib/portfile.cmake [hunk 1]
```diff
@@ -77,20 +69,6 @@ vcpkg_cmake_configure(
...
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/quickcpplib/byte.hpp" "#include \"byte/include/nonstd/byte.hpp\"" "#include <nonstd/byte.hpp>")
-endif()
-if (QUICKCPPLIB_REQUIRE_CXX20)
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/quickcpplib/span.hpp" "#ifdef QUICKCPPLIB_USE_STD_SPAN" "#if 1")
-else ()
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/quickcpplib/span.hpp" "#ifdef QUICKCPPLIB_USE_STD_SPAN" "#if 0")
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/quickcpplib/span.hpp" "#elif(_HAS_CXX20 || __cplusplus >= 202002) && __has_include(<span>)" "#elif 0")
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/quickcpplib/span.hpp" "#include \"span-lite/include/nonstd/span.hpp\"" "#include <nonstd/span.hpp>")
-endif()
-
```
Comment (BurningEnlightenment) [inline ports/ned14-internal-quickcpplib/portfile.cmake:93]: FTR the buildsystem writes these configurations to a config header file, i.e. they are burned into the installation in a buildsystem independent way (see ned14/quickcpplib#49).

**Comment by JonLiu1993** [comment]
@BurningEnlightenment, Thanks for your PR, when I test the feature by command `./vcpkg install llfio[*]` I get this error:
````
CMake Error at F:/Feature-test/lifio/vcpkg/installed/x64-windows/share/quickcpplib/quickcpplibConfig.cmake:31 (find_dependency):
  Unknown CMake command "find_dependency".
Call Stack (most recent call first):
  cmake/QuickCppLibBootstrap.cmake:32 (find_package)
  CMakeLists.txt:38 (include)
````
Please take a look.

**Comment by BurningEnlightenment** [comment]
I'll fix this upstream and report back afterwards

File: ports/llfio/portfile.cmake [hunk 2]
```diff
@@ -57,14 +57,15 @@ endif()
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
-        -DPROJECT_IS_DEPENDENCY=On
-        -Dquickcpplib_DIR=${CURRENT_INSTALLED_DIR}/share/quickcpplib
+        -Dllfio_IS_DEPENDENCY=On
+        "-DCMAKE_PREFIX_PATH=${CURRENT_INSTALLED_DIR}"
```
Comment (BurningEnlightenment) [inline ports/llfio/portfile.cmake:61]: CMake doesn't find the dependencies which are already installed without explicitly setting the prefix path for some reason. I mean it _works_, but it still seems odd. Am I missing something?

**Comment by BurningEnlightenment** [comment]
@JonLiu1993 I think this is ready for review

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
For port `llfio,ned14-internal-quickcpplib,outcome,status-code`:
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Pull request merged**